src/components/layout/Header.js

```diff
@@ -128,18 +326,60 @@ class Header extends Component {
   }
 }
 
-const mapStateToProps = ({ context, appInstance }) => {
+const mapStateToProps = ({
+  context,
+  code,
+  layout,
+  appInstanceResources,
+  appInstance,
+}) => {
+  const { userId } = context;
+  // check to see if there is already an app instance
+  // resource containing input from this user
+  const inputResource = appInstanceResources.content.find(({ user, type }) => {
+    return user === userId && type === INPUT;
+  });
+  const feedbackResource = appInstanceResources.content.find(
+    ({ user, type }) => {
+      return user === userId && type === FEEDBACK;
+    }
+  );
+  const stdinResource = appInstanceResources.content.find(({ user, type }) => {
+    return user === userId && type === STDIN;
+  });
   const {
     programmingLanguage = DEFAULT_PROGRAMMING_LANGUAGE,
   } = appInstance.content.settings;
   return {
+    userId,
+    inputResourceId: inputResource && (inputResource.id || inputResource._id),
+    stdinResourceId: stdinResource && (stdinResource.id || stdinResource._id),
+    appInstanceId: context.appInstanceId,
+    spaceId: context.spaceId,
     mode: context.mode,
     view: context.view,
+    savedCode: inputResource && inputResource.data,
+    feedback: feedbackResource && feedbackResource.data,
+    savedInput: stdinResource && stdinResource.data,
+    currentInput: code.input,
+    isInputDisplayed: layout.settings.isInputDisplayed,
```

Comment:
Usually we do not prepend `is` to booleans, so it should be `inputDisplayed`.

Reply:
DONE.